src/typing/flow_js.ml

```diff
@@ -1616,6 +1616,11 @@ let rec __flow cx (l, u) trace =
       then rec_flow cx trace (l, u)
       else rec_flow cx trace (default, u)
 
+    | (ObjTestT(reason, default, u), _) ->
```

Comment:
To be honest, I didn't have a plan here. I wrote a too small test case, saw "type is incompatible" and (poorly—typos abound) reversed the above rule without considering what the upper and lower bounds would actually be.

Is there any way to "defer" this kind of check until the upper bound is concrete?